src/transport/tcp/mod.rs

```diff
@@ -554,9 +584,16 @@ impl Stream for TcpTransport {
             };
         }
 
+        // Whenever we are receiving events by `Poll::Ready` and we choose to not propagate
+        // them to the higher levels, we should wake up the context to poll us again.
+        // Otherwise, the scheduler will not know that we are ready to be polled again.
+        let mut should_wake_up = false;
```

Comment:
I think this should be reverted, ideally either `self.listener` / `self.pending_raw_connections` / `self.pending_connections` would save the waker.
Wdyt? @dmitry-markin 🙏

Reply:
It doesn't seem we swallow any pending events from `self.listener`? As for `self.pending_raw_connections` / `self.pending_connections` we better save and trigger a waker only when new connections are added, and not in every `poll_next` call.

Reply:
Actually, I'll revert this for now and come with a follow-up. I believe this is needed and that's part of the reason why we may experience connection stability issues  🙏